src/pinns_pde_solve.jl

```diff
@@ -110,7 +135,31 @@ struct PhysicsInformedNN{isinplace,C,T,P,PH,DER,PE,AL,K} <: AbstractPINN{isinpla
         else
             _derivative = derivative
         end
-        new{iip,typeof(chain),typeof(strategy),typeof(initθ),typeof(_phi),typeof(_derivative),typeof(param_estim),typeof(additional_loss),typeof(kwargs)}(chain,strategy,initθ,_phi,_derivative,param_estim,additional_loss,kwargs)
+
+        if typeof(adaptive_loss) <: AbstractAdaptiveLoss 
+            _adaptive_loss = adaptive_loss
+        elseif adaptive_loss == nothing
+            _adaptive_loss = NonAdaptiveLoss{Float64}()
+        else
+            throw(ArgumentError("adaptive_loss must be an AbstractAdaptiveLoss or nothing"))
+        end
+
+        if log_options == nothing
+            _log_options = LogOptions()
+        else
+            _log_options = log_options
+        end
+
+        if iteration == nothing
+            _iteration = [1]
+            self_increment = true
+        else
+            _iteration = iteration
+            self_increment = false
+        end
```

Comment:
Same for these, I think just use the normal kwarg handling?

Reply:
I added LogOptions to the default kwarg handling but I think iteration still needs some initialization logic here since I need to know if the user passed in an iteration vector or not as a proxy for whether or not the adaptive reweighting loss needs to estimate its own optimization increment counter.